tests/std/tests/P0896R4_views_all/test.cpp

```diff
@@ -183,4 +210,11 @@ int main() {
         test_one(str);
         assert(ranges::equal(views::all(str), str));
     }
+
+    // Weird views, LWG-3724
+    {
+        weird_string_view v{};
+        static_assert(test_one(v));
+        test_one(v);

```

Comment:
I think this will trigger my suggested `static_assert`. If so, you can remove this test case and or put it in a new "death" test if you are really motivated. (I wouldn't refuse such a test case, but wouldn't require on either.)